zrml/swaps/src/lib.rs

```diff
@@ -733,5 +739,56 @@ mod pallet {
 
             Ok(next_pool_id)
         }
+
+        /// Returns the assets of a pool.
+        fn pool_assets(pool_id: u128) -> Result<Vec<Asset<Self::MarketId>>, DispatchError> {
+            Ok(Self::pool_by_id(pool_id)?.assets)
+        }
+
+        fn pool_status(pool_id: u128) -> Result<PoolStatus, DispatchError> {
+            Ok(Self::pool_by_id(pool_id)?.pool_status)
+        }
+
+        fn set_pool_as_inactive(pool_id: u128) -> DispatchResult {
+            Self::set_pool_status(pool_id, PoolStatus::Inactive)
+        }
+
+        fn set_pool_as_stale(
+            pool_id: u128,
+            outcome_report: &OutcomeReport,
+        ) -> Result<[usize; 2], DispatchError> {
+            let pool = Self::pool_by_id(pool_id)?;
+            let mut total_accounts: usize = 0;
+            let mut total_asset_accounts: usize = 0;
+
+            let mut assets_iter = pool.assets.into_iter();
+            let manage_asset = |asset: Asset<_>, winning_asset_idx| {
+                if let Asset::CategoricalOutcome(_, idx) = asset {
+                    if idx == winning_asset_idx {
+                        return 0;
+                    }
+                    let accounts = T::Shares::accounts_by_currency_id(asset);
+                    T::Shares::destroy_all(asset, accounts.iter().cloned());
+                    accounts.len()
+                } else {
+                    0
+                }
+            };
+
+            if let &OutcomeReport::Categorical(winning_asset_idx) = outcome_report {
+                if let Some(first_asset) = assets_iter.next() {
+                    let accounts_len = manage_asset(first_asset, winning_asset_idx);
+                    total_accounts = accounts_len;
+                    total_asset_accounts = total_asset_accounts.saturating_add(accounts_len);
+                }
+                for asset in assets_iter {
+                    let accounts_len = manage_asset(asset, winning_asset_idx);
+                    total_asset_accounts = total_asset_accounts.saturating_add(accounts_len);
+                }
+            }
+
+            Self::set_pool_status(pool_id, PoolStatus::Stale)?;
+            Ok([total_accounts, total_asset_accounts])
+        }
```

Comment:
The snippet delivers a wrong number: `total_accounts` really carries the number of accounts for `first_asset`, not the total number of accounts.